keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -324,47 +335,109 @@ def next(prompt, state, index):
 
     def generate(
         self,
-        prompt,
-        max_length,
+        inputs,
+        max_length=None,
```

Comment:
Sorry for the back and forth, but I am starting to feel give `max_length` a default to model capacity could lead to a bad UX. The vanilla GPT2 won't stop until 1024, so users would need to wait for a while to get results back. But anyway let's get this PR in, and I can open a few followups.

Reply:
Let's chat! I think we should think a little more generally and long term. How many people actually want exactly 50 unconditioned tokens for GPT2. It's kinda toy I think?

Most real workflows will usually involve fine-tuning on sequences that terminate in someway. E.g. build a summarizer, or a chatbot responder, or a translator. This get's even more true when we think about seq2seq models like T5 and BART, which will follow the same UX we are establishing here.

We want to choose a default that will scale well towards the future states of the library, and I am skeptical that requiring a `max_length` in all cases is a good idea.